### PR TITLE
Fix show-model option

### LINKED ISSUE #615: Trouble with `--show-model`
### Description

Found this while I was trying to debug an unrelated issue in `fpm` itself.

```
cd fpm
fpm build --show-model
At line 247 of file ././src/fpm_model.f90
Fortran runtime error: Index '1' of dimension 1 of array 'source%link_libraries' below lower bound of 4

Error termination. Backtrace:
#0  0x7f0a0e0f9332 in ???
#1  0x7f0a0e0f9df9 in ???
#2  0x7f0a0e0fa40a in ???
#3  0x4ad86b in info_srcfile
        at ././src/fpm_model.f90:247
#4  0x4af2fa in info_package
        at ././src/fpm_model.f90:167
#5  0x4a363c in info_model
        at ././src/fpm_model.f90:277
#6  0x4a2db9 in __fpm_model_MOD_show_model
        at ././src/fpm_model.f90:313
#7  0x40c527 in __fpm_MOD_cmd_build
        at ././src/fpm.f90:285
#8  0x4027eb in MAIN__
        at app/main.f90:67
#9  0x402d1b in main
        at app/main.f90:10
```

### Expected Behaviour

View fpm model.

### Version of fpm

9dc068d256da43085582956626881461ffdcd356

### Platform and Architecture

Linux x64, Windows x64

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by zoziha**
To close #615 .

Init the `f_source%link_libraries` array length as 0.

### CODE REVIEW

**Review by St-Maxwell — COMMENTED** [review]
It's OK.

**Review by milancurcic — APPROVED** [review]
Looks good, the issue from #615 goes away for me with this patch. Thank you!

**Comment by zoziha** [comment]
This is a relatively small correction, mainly to initialize the length of the allocatable array `f_source%link_libraries` to prevent access segfaults.
If possible, it seems that the PR can pass ~

File: src/fpm_sources.f90 [hunk 1]
```diff
@@ -97,6 +97,9 @@ subroutine add_sources_from_dir(sources,directory,scope,with_executables,recurse
 
         dir_sources(i)%unit_scope = scope
 
+        ! TODO: Add actual link_libraries ?
```
Comment (LKedward) [inline src/fpm_sources.f90:100]:
```suggestion
```
We can remove this comment since `link_libraries` are populated in `add_executable_sources` routine.

**Review by LKedward — APPROVED** [review]
Thanks for this @zoziha and many apologies for the delay in reviewing. I needed to double-check why this wasn't breaking the `link_executable` example package test case, but this all looks fine to me.

The actual `link_libraries` are populated in the [`add_executable_sources`](https://github.com/fortran-lang/fpm/blob/aec789f5a39d45d0339fc3eddc55f1f29b0099c3/src/fpm_sources.f90#L125) routine which applies any executable-specific information from the manifest (such as `link_libraries`). Your changes don't break this because of (re)allocation-on-assignment.

File: src/fpm_sources.f90 (hunk 1 above)
Comment (zoziha) [reply src/fpm_sources.f90]: Yes, it is (re)allocation-on-assignment.

**Review by LKedward — APPROVED** [review]

**Pull request merged**